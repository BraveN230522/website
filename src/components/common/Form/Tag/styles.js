import styled from '@emotion/styled';
import { css } from '@emotion/core';

const dynamicVariants = ({ variant, theme, isSelected }) =>
  ({
    light: css`
      border: solid 2px ${theme.colors.black};
      color: ${theme.colors.black};
      background: ${isSelected ? theme.colors.white : 'transparent'};

      &:hover {
        background: ${theme.colors.greyishBrown};
        color: ${theme.colors.white};

        svg {
          fill: ${theme.colors.white};
        }
      }
    `,

    dark: css`
      border: solid 2px ${theme.colors.white};
      color: ${isSelected ? theme.colors.black : theme.colors.white};
      background: ${isSelected ? theme.colors.white : theme.colors.black};

      &:hover {
        background: ${theme.colors.greyishBrown};
        color: ${theme.colors.white};

        svg {
          fill: ${theme.colors.white};
        }
      }
    `,
  }[variant]);

const TagStyles = styled.button`
  ${dynamicVariants};

  border-radius: 25px;
  cursor: pointer;
  display: flex;
  outline: none;
  font-family: ${({ theme }) => theme.fonts.DINAlternateBold};
  font-size: 14px;
  font-weight: 400;
  height: 32px;
  justify-content: space-around;
  letter-spacing: -0.1px;
  align-items: center;
  padding: 0 8px;
  text-decoration: none;
  transition: all 0.3s ease-in;
  user-select: none;
  white-space: nowrap;
  background: ${({ isSelected, theme }) => (isSelected ? theme.colors.greyishBrown : undefined)};
  color: ${({ isSelected, theme }) => (isSelected ? theme.colors.white : undefined)};
`;

export { TagStyles };
