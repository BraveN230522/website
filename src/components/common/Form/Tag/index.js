import React from 'react';
import PropTypes from 'prop-types';

import { TagStyles } from './styles';

const Tag = ({ className, children, click, isSelected, variant }) => {
  return (
    <TagStyles className={className} variant={variant} onClick={click} isSelected={isSelected}>
      {children}
    </TagStyles>
  );
};

Tag.defaultProps = {
  click: () => {},
  variant: 'light',
};

Tag.propTypes = {
  click: PropTypes.func,
  children: PropTypes.node.isRequired,
  variant: PropTypes.string,
};

export default Tag;
