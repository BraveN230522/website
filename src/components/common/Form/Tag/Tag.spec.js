import React from 'react';
import { render } from '@testing-library/react';
import withGlobals from '@swiamTestUtils/withGlobals';
import Tag from './index';

describe('<Tag />', () => {
  it('Renders a tag tag with variant light', () => {
    const { container } = render(<Tag variant="light">One tag</Tag>, {
      wrapper: withGlobals,
    });
    expect(container).toMatchSnapshot();
  });

  it('Renders a tag tag without variant', () => {
    const { container } = render(<Tag variant="light">Another tag</Tag>, {
      wrapper: withGlobals,
    });
    expect(container).toMatchSnapshot();
  });
});
