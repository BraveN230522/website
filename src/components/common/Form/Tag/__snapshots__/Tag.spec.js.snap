// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tag /> Renders a tag tag with variant light 1`] = `
.emotion-0 {
  border: solid 2px #000000;
  color: #000000;
  background: transparent;
  border-radius: 25px;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-user-selection: none;
  -moz-user-selection: none;
  -ms-user-selection: none;
  user-selection: none;
  outline: none;
  font-family: DINAlternateBold,sans-serif;
  font-size: 14px;
  font-weight: 400;
  height: 32px;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  -webkit-letter-spacing: -0.1px;
  -moz-letter-spacing: -0.1px;
  -ms-letter-spacing: -0.1px;
  letter-spacing: -0.1px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0 8px;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: all 0.3s ease-in;
  transition: all 0.3s ease-in;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.emotion-0:hover {
  background: #565656;
  color: #ffffff;
}

.emotion-0:hover svg {
  fill: #ffffff;
}

<div>
  <button
    class="emotion-0 emotion-1"
  >
    One tag
  </button>
</div>
`;

exports[`<Tag /> Renders a tag tag without variant 1`] = `
.emotion-0 {
  border: solid 2px #000000;
  color: #000000;
  background: transparent;
  border-radius: 25px;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-user-selection: none;
  -moz-user-selection: none;
  -ms-user-selection: none;
  user-selection: none;
  outline: none;
  font-family: DINAlternateBold,sans-serif;
  font-size: 14px;
  font-weight: 400;
  height: 32px;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  -webkit-letter-spacing: -0.1px;
  -moz-letter-spacing: -0.1px;
  -ms-letter-spacing: -0.1px;
  letter-spacing: -0.1px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0 8px;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: all 0.3s ease-in;
  transition: all 0.3s ease-in;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.emotion-0:hover {
  background: #565656;
  color: #ffffff;
}

.emotion-0:hover svg {
  fill: #ffffff;
}

<div>
  <button
    class="emotion-0 emotion-1"
  >
    Another tag
  </button>
</div>
`;
